### ISSUE TITLE
search results should just have a single link to the paper, or be expandable

### ISSUE THREAD

**Issue by mocxd** [GovLab/open-government-research-exchange#154]
i wrapped the search.html result cards in an `a` but didn't remove the links within the card itself due to probably sass breakage.

due to the performance problem discussed before, we can't viably have all of the deep cross-linking present in the pre-rendered cards in these ones generated by JS on the fly, and since the use cases are starting to diverge, maybe the best thing could for the result snippets to have their own template, instead of rendering out a one-off of the macro as in https://github.com/GovLab/open-government-research-exchange/blob/master/source/templates/rendered_result_item.html

if we do need deeply-linked search results, the best option in my mind is to start with a simplified result card, and then 'expand' that card by clicking on it, and perform all the mapping of the various datasets needed to drive the deep linking at the time of that click event, instead of for x number of cards loaded every time a search is performed.

**Comment by claudioccm** [comment]
ideally, all the cards would have the same appearance, to make it seamless for the users. Ideally, users should not perceive they are different things under the hood. But since we are having performance issues, let's try to edit the card template so it doesn't have all the links. I'm fine with that

**Comment by mocxd** [comment]
once concept here could be to pre-cache all of the card metadata in another json file, basically just loop through the json subsets and pull the data back in, like we are doing with the card macro during build time, then use that json file as the document store for lunr.

this would solve all the performance problems, but probably would need to be a patch 3 thing just because it is a little bit of a lift

**Comment by mocxd** [comment]
best solution for now might be to just use the card as it appears now, but wipe out all the links on the elements so it looks similar but is just one big link to the paper.

to improve UX here, we could also start linking the metadata stuff from the paper page as well, so that if they do want to click on something, all they have to do is click on the search result once and then click again on what they want from the paper page

**Comment by mocxd** [comment]
https://github.com/GovLab/open-government-research-exchange/issues/156

**Comment by claudioccm** [comment]
agreed. By pre-cache you mean loading the whole JSON to local storage, right? Technically, it could be done in a cookie, probably, but the cookie policy in europe is kinda annoying. I'd lean towards local storage, which is also 10MB, in average

**Comment by mocxd** [comment]
no i guess it is not really 'caching' technically, i just mean compile everything (including the info pulled from json subsets) into a gigantic JSON file (just for lunr) so we don't have to iterate over all the different datasets to pull in the stuff needed to hook up the links, and use that as the lunr store.  we could actually also cache search results to improve performance, but that's another discussion...

specifically, the issue is that doing what is happening (for example) on this line in the result item macro:

``` {%- set author_link = (author | returnID(authors)) ~ '-' ~ ( author | slug ) ~ '.html' -%}```

on the fly with js for half a dozen data subsets for dozens or hundreds of search cards returned every time the user types a character in the search box kills performance to the point at which the page is totally unusable (kind of like how slow/clunky the ALL page is, but even slower).  on the other hand, if all that data is in a single, flat json record per result, it should be much faster (currently we are pulling data from the papers.json records for this, and it is pretty fast).  the only time there is a problem is when we have to do what `returnID()` is doing, but on the client side, which is to iterate through all of those json subsets and return back something to build a link with.

basically we just need to use `returnID()` to pre-build a data structure that can be used by lunr during the gulp build step, rather than with js.

**Comment by mocxd** [comment]
the performance difference is that instead of firing off a single loop for each card to find the data we need to inject into the html to display the card, if we add all these cross-links in to the search result, we are firing off ~ 10 loops per card every time a card is displayed on the page.

which would be fine if we were only displaying a couple cards, if you are typing and the debounce fires with a short word, or just a common word, it will attempt to render 100s of cards on the page, which is fine with 1 loop and doesn't have a noticeable performance drag, but freezes the browser for a long time if each card has 10 loops it has to resolve before it can be displayed.
